src/test/ui/hygiene/call-site-parent-vs-expansion-parent.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-pass
+
+#![feature(decl_macro)]
+
+macro_rules! define_field {
+    () => {
+        struct S { field: u8 }
+    }
+}
+
+macro use_field($define_field: item) {
+    $define_field
+
+    // OK, both struct name `S` and field `name` resolve to definitions produced by `define_field`
+    // and living in the "root" context that is in scope at `use_field`'s def-site.
+    fn f() { S { field: 0 }; }
+}
+
+use_field!(define_field!{});
+
```

Comment:
if define_field took an identifier as an argument, and the user passed "field" as the identifier, would that have compiled before, but would fail now? Or are there other ways this is a breaking change (other than diagnostics)?

I have a feeling there could be very subtle breakage

Reply:
Yes, this *is* a subtle breakage for users of declarative macros 2.0 and their def-site hygiene.
A had hard time coming up even with one behavior-changing test though (but it turned out to fix things, rather than break).

I'll add a test for `field` passed as an identifier and `macro_rules! use_field`, but I'm pretty sure the second one is not affected by the change (it doesn't go through the "who is the parent" logic).